### DISCUSSION TITLE
[Bug] After backing up and restoring by copying the scoop folder and configuration files, scoop reset does not work on modules

### DISCUSSION THREAD

**Discussion by eidswe** [ScoopInstaller/Scoop#5409] [Help and Questions]
#### Current Behavior
After I backup and restore by copying the scoop folder and configuration files, after running scoop reset *, scoop update, scoop reset *, scoop-completion does not work, and the module must be reinstalled to use this function normally.

#### Expected Behavior
After I backup and restore by copying the scoop folder and configuration files, and run scoop reset *, scoop update, scoop reset *, I can use scoop-completion normally.

#### Additional context/output
PowerShell 7.3.1
Could not create process with command '"C:\Users\eidswe\scoop\apps\scoop-search\current\scoop-search.exe"  --hook'.
Invoke-Expression: C:\Users\eidswe\Documents\PowerShell\Microsoft.PowerShell_profile.ps1:1
Line |
   1 |  Invoke-Expression (&scoop-search --hook)
     |                    ~~~~~~~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Command' because it is null.
Import-Module: C:\Users\eidswe\Documents\PowerShell\Microsoft.PowerShell_profile.ps1:2
Line |
   2 |  Import-Module scoop-completion
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The specified module 'scoop-completion' was not loaded because no valid module file was found in any module
     | directory.
PS C:\Users\eidswe> scoop reset *
Resetting 7zip (22.01).
Linking ~\scoop\apps\7zip\current => ~\scoop\apps\7zip\22.01
Creating shim for '7z'.
Creating shim for '7zFM'.
Creating shim for '7zG'.
Creating shortcut for 7-Zip (7zFM.exe)
Persisting Codecs
Persisting Formats
Resetting adb (33.0.3).
Linking ~\scoop\apps\adb\current => ~\scoop\apps\adb\33.0.3
Creating shim for 'adb'.
Creating shim for 'dmtracedump'.
Creating shim for 'etc1tool'.
Creating shim for 'fastboot'.
Creating shim for 'hprof-conv'.
Resetting aria2 (1.36.0-1).
Linking ~\scoop\apps\aria2\current => ~\scoop\apps\aria2\1.36.0-1
Creating shim for 'aria2c'.
Resetting cpu-z-cn (2.03).
Linking ~\scoop\apps\cpu-z-cn\current => ~\scoop\apps\cpu-z-cn\2.03
Creating shortcut for CPU-Z Chinese Edition (cpuz_x64.exe)
Persisting cpuz.ini
Resetting dark (3.11.2).
Linking ~\scoop\apps\dark\current => ~\scoop\apps\dark\3.11.2
Creating shim for 'dark'.
Resetting git (2.39.1.windows.1).
Linking ~\scoop\apps\git\current => ~\scoop\apps\git\2.39.1.windows.1
Creating shim for 'sh'.
Creating shim for 'bash'.
Creating shim for 'git'.
Creating shim for 'gitk'.
Creating shim for 'git-gui'.
Creating shim for 'scalar'.
Creating shim for 'tig'.
Creating shim for 'git-bash'.
Creating shortcut for Git Bash (git-bash.exe)
Creating shortcut for Git GUI (git-gui.exe)
Resetting gpu-z (2.52.0).
Linking ~\scoop\apps\gpu-z\current => ~\scoop\apps\gpu-z\2.52.0
Creating shim for 'GPU-Z'.
Creating shortcut for GPU-Z (GPU-Z.exe)
Resetting innounp (0.50).
Linking ~\scoop\apps\innounp\current => ~\scoop\apps\innounp\0.50
Creating shim for 'innounp'.
Resetting MaaAssistantArknights (4.10.3).
Linking ~\scoop\apps\MaaAssistantArknights\current => ~\scoop\apps\MaaAssistantArknights\4.10.3
Creating shim for 'MAA'.
Creating shortcut for MAA Assistant Arknights (MAA.exe)
Get-Item: C:\Users\eidswe\scoop\apps\scoop\current\lib\install.ps1:1182
Line |
1182 |              $source = Get-Item "$dir\$source"
     |                        ~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path 'C:\Users\eidswe\scoop\apps\MaaAssistantArknights\4.10.3\gui.json' because it does not exist.
Persisting map
Persisting gui.json
Resetting openjdk (19.0.2-7).
Linking ~\scoop\apps\openjdk\current => ~\scoop\apps\openjdk\19.0.2-7
Resetting scoop-completion (0.2.4).
Linking ~\scoop\apps\scoop-completion\current => ~\scoop\apps\scoop-completion\0.2.4
Resetting scoop-search (1.2.0).
Linking ~\scoop\apps\scoop-search\current => ~\scoop\apps\scoop-search\1.2.0
Creating shim for 'scoop-search'.
Resetting steamcommunity-302 (12.1.20).
Linking ~\scoop\apps\steamcommunity-302\current => ~\scoop\apps\steamcommunity-302\12.1.20
Creating shortcut for SteamCommunity 302 (steamcommunity_302.exe)
Resetting sudo (0.2020.01.26).
Linking ~\scoop\apps\sudo\current => ~\scoop\apps\sudo\0.2020.01.26
Creating shim for 'sudo'.
Resetting SwiftSearch (7.5.1).
Linking ~\scoop\apps\SwiftSearch\current => ~\scoop\apps\SwiftSearch\7.5.1
Creating shim for 'SwiftSearch64'.
PS C:\Users\eidswe> scoop update
Updating Scoop...
Updating 'cc713_ownscoop' bucket...
Updating 'Darkatse' bucket...
Updating 'DEV-Tools' bucket...
Updating 'extras' bucket...
Updating 'java' bucket...
Updating 'main' bucket...
Updating 'nonportable' bucket...
Updating 'seumsc_scoop-seu' bucket...
Updating 'versions' bucket...
Scoop was updated successfully!
PS C:\Users\eidswe> scoop reset *
Resetting 7zip (22.01).
Linking ~\scoop\apps\7zip\current => ~\scoop\apps\7zip\22.01
Creating shim for '7z'.
Creating shim for '7zFM'.
Creating shim for '7zG'.
Creating shortcut for 7-Zip (7zFM.exe)
Persisting Codecs
Persisting Formats
Resetting adb (33.0.3).
Linking ~\scoop\apps\adb\current => ~\scoop\apps\adb\33.0.3
Creating shim for 'adb'.
Creating shim for 'dmtracedump'.
Creating shim for 'etc1tool'.
Creating shim for 'fastboot'.
Creating shim for 'hprof-conv'.
Resetting aria2 (1.36.0-1).
Linking ~\scoop\apps\aria2\current => ~\scoop\apps\aria2\1.36.0-1
Creating shim for 'aria2c'.
Resetting cpu-z-cn (2.03).
Linking ~\scoop\apps\cpu-z-cn\current => ~\scoop\apps\cpu-z-cn\2.03
Creating shortcut for CPU-Z Chinese Edition (cpuz_x64.exe)
Persisting cpuz.ini
Resetting dark (3.11.2).
Linking ~\scoop\apps\dark\current => ~\scoop\apps\dark\3.11.2
Creating shim for 'dark'.
Resetting git (2.39.1.windows.1).
Linking ~\scoop\apps\git\current => ~\scoop\apps\git\2.39.1.windows.1
Creating shim for 'sh'.
Creating shim for 'bash'.
Creating shim for 'git'.
Creating shim for 'gitk'.
Creating shim for 'git-gui'.
Creating shim for 'scalar'.
Creating shim for 'tig'.
Creating shim for 'git-bash'.
Creating shortcut for Git Bash (git-bash.exe)
Creating shortcut for Git GUI (git-gui.exe)
Resetting gpu-z (2.52.0).
Linking ~\scoop\apps\gpu-z\current => ~\scoop\apps\gpu-z\2.52.0
Creating shim for 'GPU-Z'.
Creating shortcut for GPU-Z (GPU-Z.exe)
Resetting innounp (0.50).
Linking ~\scoop\apps\innounp\current => ~\scoop\apps\innounp\0.50
Creating shim for 'innounp'.
Resetting MaaAssistantArknights (4.10.3).
Linking ~\scoop\apps\MaaAssistantArknights\current => ~\scoop\apps\MaaAssistantArknights\4.10.3
Creating shim for 'MAA'.
Creating shortcut for MAA Assistant Arknights (MAA.exe)
Persisting map
Persisting gui.json
Resetting openjdk (19.0.2-7).
Linking ~\scoop\apps\openjdk\current => ~\scoop\apps\openjdk\19.0.2-7
Resetting scoop-completion (0.2.4).
Linking ~\scoop\apps\scoop-completion\current => ~\scoop\apps\scoop-completion\0.2.4
Resetting scoop-search (1.2.0).
Linking ~\scoop\apps\scoop-search\current => ~\scoop\apps\scoop-search\1.2.0
Creating shim for 'scoop-search'.
Resetting steamcommunity-302 (12.1.20).
Linking ~\scoop\apps\steamcommunity-302\current => ~\scoop\apps\steamcommunity-302\12.1.20
Creating shortcut for SteamCommunity 302 (steamcommunity_302.exe)
Resetting sudo (0.2020.01.26).
Linking ~\scoop\apps\sudo\current => ~\scoop\apps\sudo\0.2020.01.26
Creating shim for 'sudo'.
Resetting SwiftSearch (7.5.1).
Linking ~\scoop\apps\SwiftSearch\current => ~\scoop\apps\SwiftSearch\7.5.1
Creating shim for 'SwiftSearch64'.
PS C:\Users\eidswe>

#### Possible Solution
The method of backing up and restoring directly by copying the scoop folder and configuration files does not seem to be able to copy the link. Scoop reset seems to have the effect of rebuilding links, but link rebuilding inside the "modules" folder seems to be ignored.
In addition, I hope there is a convenient way to backup the entire scoop instead of importing and exporting, so as not to need to re-download all software when restoring.

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:** 7.3.1

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

```json
//# Your configuration here
{
  "last_update": "2023-01-20T20:07:31.1035045+08:00",
  "proxy": "127.0.0.1:20171",
  "aria2-enabled": true,
  "aria2-warning-enabled": false,
  "aria2-max-connection-per-server": "16",
  "aria2-split": "32",
  "aria2-min-split-size": "1M",
  "scoop_branch": "master",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop"
}``

**Comment by rashil2000** [comment]
Can you try the workaround mentioned in https://github.com/ScoopInstaller/Scoop/discussions/5408/#discussioncomment-5121319?

**Reply by eidswe** [reply]
It's a bit tricky to reproduce this problem, so I'm not sure if the method you give works. My fix was to reinstall the module.
